### DISCUSSION TITLE
Creating a threejs and vite setup

### DISCUSSION THREAD

**Discussion by adarsh-gupta101** [thisdot/starter.dev#394] [Ideas]
I guess adding a starter kit for Three JS using vite will be a good addition to this repository

**Comment by dustinsgoodman** [comment]
Hi @adarsh-gupta101 - to date, starter.dev has been more focused on web applications. However, three.js is a very innovative project and definitely could find a place here. A question I have is how would you structure a three.js project into starter kit like our existing kits? I think this will help define the shape and structure. We'd love to help you shape the idea.

**Comment by adarsh-gupta101** [comment]
I have a rough plan for the starter kit:


1) For UI we can use React or Next or even plain HTML CSS (subjected)
2) For styling the html/ element part I prefer using tailwind CSS
3) For the threejs part we can have  the setup with:
    - a scene
    - camera
    - rendering function
    - a demo mesh(material+geometry)
    - lighting
    - and a debugging tool like dat.gui
4) for custom shaders we can have a folder for vertex/fragment shaders and import it when needed in the threejs code
5) All miscellaneous tool like prettier, eslint


In addition to above all these we can have a TypeScript support

**Reply by dustinsgoodman** [reply]
I'd love the TypeScript support to be out of the box. 😄 

I think the rest of that plan sounds great. I'd recommend Next.js with Tailwind. Kit could be called `threejs-next-tailwindcss`.

**Reply by adarsh-gupta101** [reply]
Yeah, the name seems pretty good.
